hugr-core/src/std_extensions/logic.rs

```diff
@@ -355,6 +366,9 @@ pub(crate) mod test {
     #[case(NaryLogic::And, [Some(false), None], Some(false))]
     #[case(NaryLogic::Or, [None, Some(false)], None)]
     #[case(NaryLogic::Or, [None, Some(true)], Some(true))]
+    #[case(NaryLogic::Eq, [None, Some(true), Some(true)], None)]
+    #[case(NaryLogic::Eq, [None, Some(false), Some(true)], Some(false))]
+    #[case(NaryLogic::Eq, [Some(true), Some(true), Some(true)], Some(true))]
```

Comment:
nit: the last doesn't need to be expressed in the more-complex `nary_partial_const_fold` format, right? Doing so is a test of the test framework, rather than the code-under-test! You could move it to `nary_const_fold` or remove it.